src/core/shape_inference/include/reduce_shape_inference.hpp

```diff
@@ -21,49 +21,73 @@ inline void dynamic_inference<ov::PartialShape>(const ov::PartialShape& input_sh
     output_shape = keep_dims ? ov::PartialShape::dynamic(input_shape.rank()) : ov::PartialShape::dynamic();
 }
 
-template <class T>
-void reduce_shape_infer(const ov::op::util::ReductionBase* op,
-                        bool keep_dims,
-                        const T& input_shape,
-                        T& output_shape,
-                        const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+template <class TShape>
+std::vector<TShape> reduce_shape_infer(const ov::op::util::ReductionBase* op,
+                                       bool keep_dims,
+                                       const std::vector<TShape>& input_shapes,
+                                       const ov::ITensorAccessor& tensor_accessor = ov::make_tensor_accessor()) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() >= 1);
+
+    const auto& input_shape = input_shapes[0];
     const auto& data_rank = input_shape.rank();
-    std::vector<int64_t> axes_val;
-    bool axes_are_known = get_data_as_int64<T>(1, op, axes_val, constant_data);
+    TShape output_shape;
+
+    const auto axes_val =
```

Comment:
The check for provided input shapes vector size doesn't guarantee that the number of inputs is the same. 
This is why shape infer tests with default constructor and no inputs are passing.
Also the error from  `get_input_const_data_as` should be meaningful in case of accessing input at non-existing port.
Anyway, I've updated the check to expect two shapes and validate axes rank (both inputs are required).

Reply:
The point was to provide check on top that this infer requires at least two input shapes this input shapes which indicates the call of infer is correct.
Yes the number of input shape not guarantee it. The default ctor usage is some special case and user has to be aware about it, but input shapes should be provided correctly.